tools/RepoTasks/RepoTasks.Cmdlets.Tests/AppDomainWorkerShould.cs

```diff
@@ -0,0 +1,118 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace RepoTasks.Cmdlets.Tests
+{
+    using CmdletsForTest;
+    using System;
+    using System.Collections.Generic;
+    using RemoteWorker;
+    using Xunit;
+    using System.IO;
+
+    public class AppDomainWorkerShould : IDisposable
+    {
+        private readonly AppDomain _domain;
+        private readonly IReflectionWorker _reflectionWorker;
+        private const string ExpectedAssemblyName = "RepoTasks.CmdletsForTest";
+
+        // Setup
+        public AppDomainWorkerShould()
+        {
+            var setup = new AppDomainSetup
+            {
+                ApplicationBase = AppDomain.CurrentDomain.BaseDirectory,
+            };
+
+            _domain = AppDomain.CreateDomain("AppDomainIsolation: " + Guid.NewGuid(), null, setup);
+            var type = typeof(AppDomainWorker);
+            Assert.NotNull(type.FullName);
+
+            _reflectionWorker = (IReflectionWorker)_domain.CreateInstanceFromAndUnwrap(type.Assembly.Location, type.FullName);
+        }
+
+        // TearDown
+        public void Dispose()
+        {
+            if (_domain != null) AppDomain.Unload(_domain);
+        }
+
+        [Fact]
```

Comment:
Use 

```[Trait(Category.AcceptanceType, Category.CheckIn)]```  for each of these tests, this will make them run in the check-in build.  Also, please add this path and assembly to the test mappings file:https://github.com/Azure/azure-powershell/blob/preview/TestMappings.json

Reply:
@markcowl 
The project that owns the Category class with the constants (AcceptanceType, CheckIn) is dependent on many others.  Is it a good idea just to copy paste the requires constants instead of adding all the dependencies?